third_party/xla_client/xrt_computation_client.cc

```diff
@@ -1083,7 +1083,16 @@ std::unique_ptr<xrt::XLAComputation> XrtComputationClient::CreateXrtComputation(
   if (devices.size() > 1) {
     auto device_assignment = config->mutable_device_assignment();
     auto computation_device = device_assignment->add_computation_devices();
-    for (int64_t i = 0; i < devices.size(); ++i) {
+    auto local_device = GetLocalDevices();
+    int64_t local_tpu;
+    for (auto& device : local_device) {
+      if (device.compare(0, 4, "CPU:") == 0 && devices.size() > 1) {
+        continue;
+      }
+      local_tpu = std::stoi(device.substr(4));
+    }
+    int64_t local_ordinal_start = 32 * (local_tpu / 32);
```

Comment:
make it const

Reply:
Also we can't use 32 here. It at least needs to be an env var because other hardware will not necessarily have 32 devices per node.